src/cms-components/core/container.js

```diff
@@ -17,38 +17,66 @@
 import React from 'react';
 import CmsContainerItem from './container-item';
 import { addBeginComment, addEndComment } from '../../utils/add-html-comment';
-import { PreviewContext } from '../../context';
+import { ComponentDefinitionsContext, PageModelContext, PreviewContext } from '../../context';
 
 export default class CmsContainer extends React.Component {
-  renderContainerWrapper(configuration, preview) {
+  renderContainerWrapper(configuration, pageModel, preview, componentDefinitions) {
     if (preview) {
       return (
         // need to wrap container inside a div instead of React.Fragment because otherwise HTML comments are not removed
         <div>
-          <div className="hst-container"
-               ref={(containerElm) => { this.addMetaData(containerElm, configuration, preview); }}>
-            { this.renderContainer(configuration) }
+          <div
+            className="hst-container"
+            ref={containerElm => {
+              this.addMetaData(containerElm, configuration, preview);
+            }}
+          >
+            {this.renderContainer(configuration, pageModel, preview, componentDefinitions)}
           </div>
         </div>
       );
     } else {
       return (
         <React.Fragment>
-          { this.renderContainer(configuration) }
+          {this.renderContainer(configuration, pageModel, preview, componentDefinitions)}
         </React.Fragment>
       );
     }
   }
 
-  renderContainer(configuration = { components: [] }) {
-    if (configuration.components && configuration.components.length > 0) {
-      // render all of the container-item-components
-      return configuration.components.map((component) => {
-        return (
-          <CmsContainerItem configuration={component} key={component.id} />
-        );
-      });
+  renderContainer(configuration = { components: [] }, pageModel, preview, componentDefinitions) {
+    const { components, label } = configuration;
+
+    // don't render anything when there're no components found
+    if (!components || components.length === 0) {
+      return;
+    }
+
+    // get component item components
+    const containerItemComponents = components.map(component => {
+      return <CmsContainerItem configuration={component} key={component.id} />;
+    });
+
+    // check if component container is found in component definitions
+    const ContainerComponent = componentDefinitions[label] && componentDefinitions[label].component;
+
+    // if found then wrap container items with this component
+    if (!!ContainerComponent) {
```

Comment:
forgot to remove the props

Reply:
As far as I understand the purpose of `renderContainer` as a function is to render a container component which is done by calling `React.createElement(ContainerComponent, ...)`. That should be the main part of this function, but here it stays under if statement.
Can we swap those return statements to be something like that?

```javascript
if (!ContainerComponent) {
  return containerItemComponents;
}

return React.createElement(
  ContainerComponent,
  { configuration, pageModel, preview, componentDefinitions },
  containerItemComponents
);
```

In this case, rendering `containerItemComponents` will be more like an edge case from a readability point of view.

Reply:
Good notice! Yes you can swap them around.